### PR TITLE
Creating Configuration Forms with DDM Annotations | LRDOCS-4635

### PR DESCRIPTION
**Pull request by rbohl**
https://issues.liferay.com/browse/LRDOCS-4635, I decided to kick this off after Ray Auge's post in the documentation slack channel. It's been in the queue for a while, so I figured I'd use this reminder to get it started. If it's useless and all wrong, we can definitely start another draft.

### CODE REVIEW

File: en/developer/tutorials/code/guestbook/08-search/com-liferay-docs-guestbook/modules/guestbook/guestbook-web/src/main/java/com/liferay/docs/guestbook/portlet/GuestbookPortlet.java [hunk 1]
```diff
@@ -1,7 +1,10 @@
 package com.liferay.docs.guestbook.portlet;
```
Comment (pei-jung) [inline en/developer/tutorials/code/guestbook/08-search/com-liferay-docs-guestbook/modules/guestbook/guestbook-web/src/main/java/com/liferay/docs/guestbook/portlet/GuestbookPortlet.java:1]: This doesn't look related. Is it added by mistake?

File: en/developer/frameworks/articles/configuration/08-customizing-the-ui/03-configuration-forms.markdown [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+[`UserFileUploadsConfiguration.java`](https://github.com/liferay/liferay-portal/blob/7.2.0-ga1/modules/apps/users-admin/users-admin-api/src/main/java/com/liferay/users/admin/configuration/UserFileUploadsConfiguration.java)
+
+[`UserFileUploadsConfigurationBeanDeclaration.java`](https://github.com/liferay/liferay-portal/blob/7.2.0-ga1/modules/apps/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/internal/configuration/definition/UserFileUploadsConfigurationBeanDeclaration.java)
+
+[`UserFileUploadsConfigurationDDMFormDeclaration.java`](https://github.com/liferay/liferay-portal/blob/7.2.0-ga1/modules/apps/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/internal/configuration/definition/UserFileUploadsConfigurationDDMFormDeclaration.java)
+
+## Additional Details
+
+Note: we don't have docs for the Forms DDM annotations (I know because I write
```
Comment (pei-jung) [inline en/developer/frameworks/articles/configuration/08-customizing-the-ui/03-configuration-forms.markdown]: Is this not the docs for the DDM annotation? https://portal.liferay.dev/docs/7-1/tutorials/-/knowledge_base/t/adding-settings-to-form-field-types

File: en/developer/frameworks/articles/configuration/08-customizing-the-ui/03-configuration-forms.markdown [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    starts with the next step.
+
+4.  Write a `ConfigurationForm` class, including just the fields that you want
+    to leverage the enhanced forms capability. This will look similar to the
+    configuration interface, but with field annotations from the Liferay
+    [Dynamic Data Mapping API](https://github.com/liferay/liferay-portal/tree/7.2.0-ga1/modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/annotations) 
+    rather than the bndtools metatype specification.
+
+5.  Implement a [`ConfigurationDDMFormDeclaration`](https://github.com/liferay/liferay-portal/blob/7.2.0-ga1/modules/apps/configuration-admin/configuration-admin-api/src/main/java/com/liferay/configuration/admin/definition/ConfigurationDDMFormDeclaration.java) 
+    to mark your configuration as having a `ConfigurationForm`.
```
Comment (pei-jung) [inline en/developer/frameworks/articles/configuration/08-customizing-the-ui/03-configuration-forms.markdown:26]: Make sure to include the `configurationPid` in the component property so they can be wired properly.

File: en/developer/frameworks/articles/configuration/08-customizing-the-ui/03-configuration-forms.markdown [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+1.  [Create an OSGi module](/docs/7-2/reference/-/knowledge_base/r/creating-a-project).
+
+2.  Configure the dependencies.
+
+3.  Write the 
+    [configuration interface](/docs/7-2/frameworks/-/knowledge_base/f/creating-a-configuration-interface) 
+    as usual. This step is covered elsewhere, so the real code example here
+    starts with the next step.
+
+4.  Write a `ConfigurationForm` class, including just the fields that you want
```
Comment (pei-jung) [inline en/developer/frameworks/articles/configuration/08-customizing-the-ui/03-configuration-forms.markdown]: Note that the fields need to match whatever is defined in the configuration interface

**Review by pei-jung — COMMENTED** [review]
This looks great. Thanks for working on this!

File: en/developer/frameworks/articles/configuration/08-customizing-the-ui/03-configuration-forms.markdown (hunk 2 above)
Comment (rbohl) [reply en/developer/frameworks/articles/configuration/08-customizing-the-ui/03-configuration-forms.markdown]: Well, that's what we included in 7.1, but we should have more, and we should have it for 7.2. But we don't need it here, since that's not what we're showing.

File: en/developer/tutorials/code/guestbook/08-search/com-liferay-docs-guestbook/modules/guestbook/guestbook-web/src/main/java/com/liferay/docs/guestbook/portlet/GuestbookPortlet.java (hunk 1 above)
Comment (rbohl) [reply en/developer/tutorials/code/guestbook/08-search/com-liferay-docs-guestbook/modules/guestbook/guestbook-web/src/main/java/com/liferay/docs/guestbook/portlet/GuestbookPortlet.java:1]: Yes! Related to another task. I was wondering how this code disappeared :)

**Comment by rbohl** [comment]
Sente here: https://github.com/sez11a/liferay-docs/pull/4629
